src/main/java/seedu/address/model/loan/Nric.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Class that stores the NRIC of a person.
+ */
+public class Nric {
+
+    public static final String MESSAGE_NRIC_CONSTRAINTS =
+            "Nric should be singapore issued. It may be blank. ";
+
+    public static final String NRIC_VALIDATION_REGEX = "^[STGF][0-9]{7}[A-Z]$";
```

Comment:
Different NRIC start digits have different check sums.

I propose changing the regex to `^[ST]\d{7}[A-JZ]|[FG]\d{7}[K-NPQRTUWX]$`

Reply:
Just a warning, the '\' above should be '\\'. Java will read '\\' as a single backslash.

Reply:
Incorrect. `\d` is the regex to match a digit. If you do `\\` it will match a single backslash, which is not what we want.